src/units/abilities.cpp

```diff
@@ -1271,18 +1271,18 @@ unit_ability_list attack_type::get_weapon_ability(const std::string& ability) co
 	}
 
 	abil_list.append(abil_other_list);
-	if(!abil_list.empty()){
-		abil_list = overwrite_special_checking(ability, abil_list, abil_list, "filter_student");
-	}
```

Comment:
Just wondering if moving this line could add a bug in udisplay.cpp's `unit_attack` function.

Reply:
if a leadershiptype abilitiy is overwrited, the animation willbe played then what not actually but i will cancel this change

Reply:
Should I review this PR now, or wait for another commit to be pushed to it?

Reply:
Now